docs/conf.py

```diff
@@ -536,7 +536,7 @@
 
 # This is processed by Jinja2 and inserted before each notebook
 nbsphinx_prolog = r"""
-{% set docname = 'docs/' + env.doc2path(env.docname, base=None) %}
+{% set docname = 'docs/' + env.doc2path(env.docname, base=None) | string %}
```

Comment:
For `sphinx>=8.0` [`BuildEnvironment.doc2path()`](https://www.sphinx-doc.org/en/master/extdev/envapi.html#sphinx.environment.BuildEnvironment.doc2path) started returning a `_StrPath` object, which resulted in the following exception...

![image](https://github.com/user-attachments/assets/2879f63a-f374-459e-bb8b-f81c2d584c03)

To fix this I added `| string` to the `docname` declaration in `nbsphinx_prolog`.  This converts the composed `docname` into a string.